singularity-eos/eos/eos_base.hpp

```diff
@@ -34,6 +34,29 @@ static inline auto member_func_name(const char *type_name, const char *func_name
 namespace singularity {
 namespace eos_base {
 
+namespace impl {
+// Cuda doesn't have strcat, so we implement it ourselves
+PORTABLE_FORCEINLINE_FUNCTION
+char *StrCat(char *destination, const char *source) {
+  int i, j;
+
+  // move to the end of the destination string
+  // specifically avoid strlen, which isn't on GPU
+  for (i = 0; destination[i] != '\0'; i++)
+    ;
```

Comment:
`clang-format` what are you doing? Should I disable clang-format here or add an empty `{}` to short-circuit this weird newline thing?

Reply:
I don't mind this too much since it very clearly identifies this as a `for` loop with no function other than to increment. The empty braces would also communicate this, so that's fine with me as well.